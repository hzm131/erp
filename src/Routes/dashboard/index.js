import React from 'react';
import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import(/* webpackChunkName: "Dashboard" */ './Dashboard'),
  loading: ()=><div></div>,
});
