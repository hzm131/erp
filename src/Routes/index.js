import React, { Component } from 'react'
import { Router } from '@reach/router'
import Loadable from 'react-loadable'

import { LocaleProvider } from 'antd'
import zh_CN from 'antd/lib/locale-provider/zh_CN'
import 'moment/locale/zh-cn'

// routes
import POS from './pos'
import Dashboard from './dashboard'
import Login from './login'
import OldPosSystem from './old_pos_system'

const NotFound = Loadable({
  loader: () => import(/* webpackChunkName: "NotFound" */ 'components/NotFound/NotFound'),
  loading: () => <div />
})

let basepath = '/'
if (window.localStorage) {
  basepath = window.localStorage.basepath || '/'
}

export default class extends Component {
  render() {
    return (
      <LocaleProvider locale={zh_CN}>
        <Router basepath={basepath}>
          <NotFound default />
          <Login path="/" />
          <Dashboard path="dashboard/*" />
          <POS path="pos" />
          <OldPosSystem path="old_pos_system" />
        </Router>
      </LocaleProvider>
    )
  }
}
