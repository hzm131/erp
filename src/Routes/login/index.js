import React from 'react'
import Loadable from 'react-loadable'

export default Loadable({
  loader: () => import(/* webpackChunkName: "Login" */ './Login'),
  loading: () => <div />
})
