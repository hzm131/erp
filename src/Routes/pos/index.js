import React from 'react'
import Loadable from 'react-loadable'

export default Loadable({
  loader: () => import(/* webpackChunkName: "Login" */ './pos_system'),
  loading: () => <div />
})
