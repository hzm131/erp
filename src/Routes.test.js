import React from 'react'
import Routes from './Routes'

import { shallow, mount } from 'enzyme'

import {
  createHistory,
  createMemorySource,
  LocationProvider
} from '@reach/router'

const App = () => (
  <LocationProvider history={createHistory(createMemorySource('/'))}>
    <Routes />
  </LocationProvider>
)

it('renders without crashing using shallow', () => {
  const shallowed = shallow(<App />)
  expect(shallowed).toMatchSnapshot()
})

it('renders without crashing using mount', () => {
  const mounted = mount(<App />)
  expect(mounted).toMatchSnapshot()
})
