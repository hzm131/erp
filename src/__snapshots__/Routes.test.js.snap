// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing using mount 1`] = `
<div
  role="group"
  style={
    Object {
      "outline": "none",
    }
  }
  tabIndex="-1"
>
  <div />
</div>
`;

exports[`renders without crashing using shallow 1`] = `
<LocationProvider
  history={
    Object {
      "_onTransitionComplete": [Function],
      "listen": [Function],
      "location": Object {
        "key": "initial",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "navigate": [Function],
      "transitioning": false,
    }
  }
>
  <Component />
</LocationProvider>
`;
